numpy/lib/npyio.py

```diff
@@ -728,20 +728,20 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
     zipf.close()
 
 
-def _getconv(dtype):
-    """ Find the correct dtype converter. Adapted from matplotlib """
+def _floatconv(x):
+    try:
+        return float(x)  # The fastest path.
+    except ValueError:
+        if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
+            try:
+                return float.fromhex(x)
+            except ValueError:
+                pass
+        raise  # Raise the original exception, which makes more sense.
 
-    def floatconv(x):
-        try:
-            return float(x)  # The fastest path.
-        except ValueError:
-            if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
-                try:
-                    return float.fromhex(x)
-                except ValueError:
-                    pass
-            raise  # Raise the original exception, which makes more sense.
 
+def _getconv(dtype):
+    """ Find the correct dtype converter. Adapted from matplotlib """
```

Comment:
Can you  add a note here that where possible the returned converter should be equality-comparable to enable optimizations?

Reply:
done